lib/rules/await-async-query.js

```diff
@@ -101,3 +105,18 @@ function isPromiseResolved(node) {
   // promise.then(...)
   return hasAThenProperty(parent);
 }
+
+function hasClosestExpectResolvesRejects(node) {
+  if (!node.parent) {
```

Comment:
Is this necessary? Because on L34 the function is called with `isAwaited(node.parent.parent)`, which implies that `node.parent` cannot be `undefined`.

Reply:
Good catch. I initially wrote a utility function called `findClosestExpectCall`that was supposed to be reusable. At that time the `node.parent` thing was necessary because the function was recursive. Then I refactored it and indeed, it's not necessary.

Reply:
I tried to remove the `node.parent` check and test broke which is, in the end, logical. While `isAwaited(node.parent.parent)` implies `node.parent` cannot be `undefined`, it's very possible that `hasClosestExpectResolvesRejects` goes beyond `node.parent.parent` (until the `Program` node) in case you're not using an `expect` statement.